### PR TITLE
Port of Advanced Plane from Gazebo Classic to Gazebo

### PR DESCRIPTION
**Pull request by frede791**
### Port
The advanced plane existed in old Gazebo classic and relies on the advanced liftdrag plugin. This port enables the user to fly the advanced plane in new Gazebo as well. The physics of the plane have not been modified and all required parameters have been taken from the original file. I have also submitted a PR for the advanced liftdrag plugin upstream in gz-sim (https://github.com/gazebosim/gz-sim/pull/2185). Without this plugin, the plane will not fly. To test this before integration, one can get plugin files from the other PR, place them in the necessary directories (as shown in the PR), compile them and place the generated .so files in `.gz/sim/plugins`. This will allow you to fly the advanced plane in Gazebo through QGC or AMC. 

### Test coverage
Flew a test course with tight turns, ascends and loitering patterns. I saw no unusual behaviour. 
- Simulation/hardware testing logs: https://review.px4.io/plot_app?log=78a4f0e2-80f5-4165-96da-c18c544c9444

### CODE REVIEW

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
Isn't the meshfile actually identical to that of the normal plane?

I don't think we need the full meshfile in tree

File: ROMFS/px4fmu_common/init.d-posix/airframes/4004_gz_standard_vtol [hunk 1]
```diff
@@ -76,7 +76,6 @@ param set-default CA_SV_CS1_TRQ_R 0.5
 param set-default CA_SV_CS2_TYPE 3
 param set-default CA_SV_CS2_TRQ_P 1.0
 
-param set-default FW_L1_PERIOD 12
```
[the remark this answers is not in the document]
Comment (frede791) [reply ROMFS/px4fmu_common/init.d-posix/airframes/4004_gz_standard_vtol:79]: No but it is no longer being used by the standard VTOL. Will remove it and can create a separate PR.

**Comment by frede791** [comment]
In theory we don't need the meshes, but the "regular" plane currently does not exist in new gazebo, so I think it is actually needed.  I could maybe specify a file path to the gz-classic repo but I think this is cleaner. If I port the regular plane as well, then I can change the mesh path and set them to the same one for both.

**Comment by Jaeyoung-Lim** [comment]
> In theory we don't need the meshes, but the "regular" plane currently does not exist in new gazebo, so I think it is actually needed. I could maybe specify a file path to the gz-classic repo but I think this is cleaner. If I port the regular plane as well, then I can change the mesh path and set them to the same one for both.

Here is where the "normal" plane is: https://github.com/PX4/PX4-Autopilot/tree/main/Tools/simulation/gz/models/rc_cessna

File: ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane [hunk 2]
```diff
@@ -0,0 +1,87 @@
+#!/bin/sh
+#
+# @name Advanced Plane SITL
+#
+
+. ${R}etc/init.d/rc.fw_defaults
+
+PX4_SIMULATOR=${PX4_SIMULATOR:=gz}
+PX4_GZ_WORLD=${PX4_GZ_WORLD:=default}
+PX4_SIM_MODEL=${PX4_SIM_MODEL:=Advanced_Plane}
```
Comment (Jaeyoung-Lim) [inline ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane:10]: Shouldn't this be advanced_plane?

File: ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+param set-default SIM_GZ_EN 1
+
+param set-default SENS_EN_GPSSIM 1
+param set-default SENS_EN_BAROSIM 0
+param set-default SENS_EN_MAGSIM 1
+param set-default SENS_EN_ARSPDSIM 1
```
Comment (Jaeyoung-Lim) [inline ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane:17]: You have an airspeed sensor attached in gazebo, so you don't need a simulated airspeed sensor

File: ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+param set-default SIM_GZ_SV_FUNC1 201
+param set-default SIM_GZ_SV_FUNC2 202
+param set-default SIM_GZ_SV_FUNC3 203
+param set-default SIM_GZ_SV_FUNC4 204
+param set-default SIM_GZ_SV_FUNC5 205
+param set-default SIM_GZ_SV_FUNC6 206
+
+
+set MIXER_FILE etc/mixers-sitl/plane_sitl.main.mix
+set MIXER custom
```
Comment (Jaeyoung-Lim) [inline ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane:87]: We should remove the mixer, this has been deprecated

File: ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+. ${R}etc/init.d/rc.fw_defaults
+
+PX4_SIMULATOR=${PX4_SIMULATOR:=gz}
+PX4_GZ_WORLD=${PX4_GZ_WORLD:=default}
+PX4_SIM_MODEL=${PX4_SIM_MODEL:=Advanced_Plane}
+
+param set-default SIM_GZ_EN 1
+
+param set-default SENS_EN_GPSSIM 1
+param set-default SENS_EN_BAROSIM 0
```
Comment (Jaeyoung-Lim) [inline ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane:15]: I think the default is already zero, so no need to set the default to zero again

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
I think we anyway need to wait for the gz PR to be merged since the plugin is on the gz side.

We can try to hold the plugin px4 side in the meantime and then deprecate it later if you want to get this in quicker

- Also, please cleanup the commit history since there are large mesh files along the commits

File: ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane (hunk 3 above)
Comment (frede791) [reply ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane:17]: Screenshot from 2023-10-03 09-40-23

The simulation complains if the airspeed sensor is not defined in the airframe file

File: ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane (hunk 2 above)
Comment (frede791) [reply ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane]: either one works

File: ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane (hunk 3 above)
Comment (Jaeyoung-Lim) [reply ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane:16]: Ah, it was the air pressure sensor -> can we use the gazebo simulated airspeed sensor?

Comment (frede791) [reply ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane:16]: Hmm maybe? I tried placing the air speed sensor in the sdf file and took out the ARSPDSIM from the airframe but it then again complains about the invalid airspeed. However, maybe I just defined it wrong in the sdf file? Where can I find the documentation for the sensor (not the class reference but the description of what variables can be written into the sdf file).

**Comment by frede791** [comment]
https://github.com/gazebosim/gz-sim/pull/2185 has been merged into gazebo. Advanced can now be used in PX4.

**Comment by Jaeyoung-Lim** [comment]
@frede791 Is it also released? Or when would it be part of the release?

**Comment by frede791** [comment]
@Jaeyoung-Lim They are currently porting 7 -> 8 and I would think it would be released as part of that. https://github.com/gazebosim/gz-sim/pull/2224

**Comment by Jaeyoung-Lim** [comment]
@frede791 Could you mention this in the documentation? Otherwise, LGTM!

**Review by Jaeyoung-Lim — APPROVED** [review]

File: ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane (hunk 3 above)
Comment (Jaeyoung-Lim) [reply ROMFS/px4fmu_common/init.d-posix/airframes/4008_gz_advanced_plane:16]: Lets address this in a followup

File: Tools/simulation/gz/models/advanced_plane/model.sdf [hunk 6]
```diff
@@ -286,7 +286,7 @@
         <geometry>
           <mesh>
             <scale>0.1 0.1 0.1</scale>
-            <uri>model://rc_cessna/meshes/rudder.dae</uri>
+            <uri>https://fuel.gazebosim.org/1.0/frede791/models/Advanced%20Plane/tip/files/meshes/rudder.dae</uri>
```
Comment (Jaeyoung-Lim) [inline Tools/simulation/gz/models/advanced_plane/model.sdf:289]: Good for now, but would like to put this under the PX4 org in ignition fuel

Comment (frede791) [reply Tools/simulation/gz/models/advanced_plane/model.sdf:289]: Yes I talked with @dagar about this. He added me as a member to the fuel PX4 team on Friday so I will make a PR to adjust all of these.

**Comment by Jaeyoung-Lim** [comment]
> @Jaeyoung-Lim https://github.com/PX4/PX4-user_guide/pull/2843
The PR for the documentation is ready to go.

I meant more the fact that it is not available yet in the released gz distribution.

**Pull request merged**